doc/source/serve/production-guide/best-practices.md

```diff
@@ -91,4 +99,26 @@ deployment_statuses:
   message: ''
 ```
 
-For Kubernetes deployments with KubeRay, tighter integrations of `serve status` with Kubernetes are available. See [Getting the status of Serve applications in Kubernetes](serve-getting-status-kubernetes).
\ No newline at end of file
+For Kubernetes deployments with KubeRay, tighter integrations of `serve status` with Kubernetes are available. See [Getting the status of Serve applications in Kubernetes](serve-getting-status-kubernetes).
+
+## Best practices for HTTP requests
+
+Most examples in these docs use straightforward `get` or `post` requests using Python's `requests` library, such as:
+
+```{literalinclude} ../doc_code/requests_best_practices.py
+:start-after: __prototype_code_start__
+:end-before: __prototype_code_end__
+:language: python
+```
+
+This pattern is useful for prototyping, but it isn't sufficient for production. In production, HTTP requests should use:
+
+* Retries: requests may occasionally fail due to transient issues (e.g. slow network, node failure, power outage, spike in traffic, etc.). Retry failed requests a handful (~10) of times to account for these issues.
+* Exponential backoff: To avoid bombarding the Serve application with retries during a transient error, apply an exponential backoff on failure. Each retry should wait exponentially longer than the previous one before running. For example, the first retry may happen 0.1 seconds after a failure, the next one 0.4s (4 x 0.1), then 1.6s, 6.4s, 25.6s, etc.
```

Comment:
```suggestion
* Exponential backoff: To avoid bombarding the Serve application with retries during a transient error, apply an exponential backoff on failure. Each retry should wait exponentially longer than the previous one before running. For example, the first retry may happen 0.1s after a failure, and subsequent retries happen 0.4s (4 x 0.1), 1.6s, 6.4s, 25.6s, etc. after the failure.
```